### PR TITLE
New stepfunction workflow - Serverlessly generate video content summary with Amazon Bedrock

### PR DESCRIPTION
**Pull request by fanhongy**
*Issue #326 *

*Description of changes:*

In this use case, users upload meeting recordings via a UI. This UI is hosted on an S3 bucket and delivered through Amazon CloudFront. When a video is uploaded, it triggers an AWS account event. Amazon EventBridge captures this event. The system either transcodes the video for Amazon Transcribe or directly feeds it if already compatible.

Amazon Transcribe then creates a transcription of the video. This transcription serves as the input for an Amazon Bedrock API request. We use the Jurassic-2 Ultra model from AI21 Lab to summarize the transcript. The summary is stored in DynamoDB. Users can access it on the UI using the video ID.

Additionally, there's an error-handling component. An AWS Lambda function handles errors. It's triggered by Amazon EventBridge rules. These rules detect failures from AWS services, like Amazon MediaConvert failing to convert a video.
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jainakhi** [comment]
The system either transcodes the video for Amazon Transcribe or directly feeds it if already compatible to the StepFunctions State Machine.

The State Machine then uses Amazon Transcribe to create a transcription of the video with automatic language detection. This transcription serves as the input for an Amazon Bedrock API request.

**Comment by bls20AWS** [comment]
There are over 2k files in this PR, please remove these packages from the PR.

**Comment by fanhongy** [comment]
Hey @bls20AWS thanks for check the PR. As per I checked most files are from boto3 and botocore Python lib as required for the lambda function to execute Bedrock SDK: 
```
.gz 391
.json 1209
.py 203
.pyc 194
```

Can you please ignore most of those files and merge the PR? I can remove the lambda function layer once the latest boto3 version is general available in lambda Python runtime.

**Comment by bls20AWS** [comment]
You need to remove the files from the repo and add them to the package.json or the requirements.txt as dependancies.  they do not need to be committed to the repository.

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+    "title": "Serverlessly generate video content summary with Amazon Bedrock",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:2]:
```suggestion
    "title": "Generate video content summary with Amazon Bedrock",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "description": "Meeting recordings uploaded through a UI trigger an AWS event, leading to transcription by Amazon Transcribe and summarization using AI21 Lab's Jurassic-2 Ultra model, with results stored in DynamoDB and accessible via the UI.",
+    "language": "Typescript,Python",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:4]:
```suggestion
    "language": "Python",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    "usecase": "",
+    "type": "Standard",
+    "diagram": [
+        "/image/bedrock-transcribe-workfloww.png"
+    ],
+    "videoId": "",
+    "level": "300",
+    "framework": "CDK",
+    "services": [
+        "Amazon Bedrock",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:15]:
```suggestion
        "bedrock",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        "AWS Step Functions",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:16]:
```suggestion
        "sfn",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        "Amazon CloudFront",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:17]: cloudfront

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        "Amazon Transcribe"
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:18]:
```suggestion
        "transcribe"
```

File: serverless-bedrock-video-content-summary/example-workflow.json (hunk 5 above)
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:17]:
```suggestion
        "cloudfront",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    ],
+    "introBox": {
+        "headline": "How it works",
+        "text": [
+            "The step function for the transcoding workflow orchestrates the AWS Elemental MediaConvert service process:",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:23]:
```suggestion
            "The AWS Step Functions state machine orchestrates the transcoding for videos using AWS Elemental MediaConvert. Users upload videos through a user-friendly interface hosted on an S3 bucket and delivered via Amazon CloudFront. When a video is uploaded, an event in the AWS account triggers Amazon EventBridge, capturing the event. The system then proceeds to transcode the video for Amazon Transcribe or directly feeds it if it's already compatible. Both the transcoding and Amazon Bedrock workflow are orchestrated by the AWS Step Functions state machine. For additional information, please refer to the GitHub repository.",
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+            "In this use case, users upload a video via a UI. This UI is hosted on an S3 bucket and delivered through Amazon CloudFront.",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:24]:
```suggestion

```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            "When a video is uploaded, it triggers an AWS account event. Amazon EventBridge captures this event. The system either transcodes the video for Amazon Transcribe or directly feeds it if already compatible.",
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:25]:
```suggestion

```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+            "Both trancode and bedrock workflow are orchestrated by AWS Step Functions state machine, check the Github repo for more information"
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:26]:
```suggestion

```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+                "headline": "",
+                "payloadURL": ""
+            }
+        ]
+    },
+    "resources": {
+        "headline": "Additional Resources",
+        "bullets": [
+            "https://aws.amazon.com/bedrock/titan/?gclid=CjwKCAiAsIGrBhAAEiwAEzMlC2GSrdSUf9pxrcjBmLL6Nwv4dGrEZcGQTvJzVQldKys9om0UJRENmxoCLbcQAvD_BwE&trk=fc80ff15-2333-44d6-b942-ac55ffa29b7d&sc_channel=ps&ef_id=CjwKCAiAsIGrBhAAEiwAEzMlC2GSrdSUf9pxrcjBmLL6Nwv4dGrEZcGQTvJzVQldKys9om0UJRENmxoCLbcQAvD_BwE:G:s&s_kwcid=AL!4422!3!655649172381!e!!g!!amazon%20bedrock!19966640379!149336804778",
+            "https://aws.amazon.com/pm/transcribe/?gclid=CjwKCAiAsIGrBhAAEiwAEzMlC3wUxDxX2S--25-QEcVjoLRVWzAFHufxH9x0hhxKxelgGU6jSAOSzBoCf6cQAvD_BwE&trk=5201f5f7-bf8b-4a2d-b097-1e900426c1d2&sc_channel=ps&ef_id=CjwKCAiAsIGrBhAAEiwAEzMlC3wUxDxX2S--25-QEcVjoLRVWzAFHufxH9x0hhxKxelgGU6jSAOSzBoCf6cQAvD_BwE:G:s&s_kwcid=AL!4422!3!652937811575!e!!g!!amazon%20transcribe!19909696445!152804588132"
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:57]:
```suggestion
"link:"https://aws.amazon.com/bedrock/titan/"
```

File: serverless-bedrock-video-content-summary/example-workflow.json [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        "payloads": [
+            {
+                "headline": "",
+                "payloadURL": ""
+            }
+        ]
+    },
+    "resources": {
+        "headline": "Additional Resources",
+        "bullets": [
```
Comment (bls20AWS) [inline serverless-bedrock-video-content-summary/example-workflow.json:52]:
```suggestion
        "bullets": [{
```

**Comment by bls20AWS** [comment]
This is missing an .asl file that describes the workflow and also please specify which is the templateFile file  location.

The workflow will not go live without these 2 components

"templateFile": ",
"ASL": ""

**Pull request merged**